client/taskenv/env.go

```diff
@@ -730,6 +747,31 @@ func buildPortEnv(envMap map[string]string, p structs.Port, ip string, driverNet
 	}
 }
 
+// SetUpstreams defined by connect enabled group services
+func (b *Builder) AddUpstreams(upstreams []structs.ConsulUpstream) *Builder {
+	b.mu.Lock()
+	b.upstreams = append(b.upstreams, upstreams...)
```

Comment:
The one thing I'm not sure about (and probably left undone and untested in the preview :disappointed:) is whether or not we should append or set here. Since services are one of the few things supported by inplace updates I think we want to `Set` instead of `Add`. Looking at how AR and TR Update methods handle `taskenv.Builder` should answer it.

Although since no interpolation or env vars affect running tasks, it really only matters during task restarts. So as long as we build the env from scratch on restarts, it probably doesn't matter what happens (if anything) to the env during AR/TR Updates.